src/org/zaproxy/zap/extension/autoupdate/ManageAddOnsDialog.java

```diff
@@ -425,13 +432,26 @@ public void valueChanged(ListSelectionEvent e) {
 				}});
 			
 			uninstalledAddOnsTable.setModel(uninstalledAddOnsModel);
-			uninstalledAddOnsTable.getColumnModel().getColumn(0).setMaxWidth(20);
-			uninstalledAddOnsTable.getColumnModel().getColumn(1).setPreferredWidth(50);
-			uninstalledAddOnsTable.getColumnModel().getColumn(2).setPreferredWidth(150);
-			uninstalledAddOnsTable.getColumnModel().getColumn(3).setPreferredWidth(300);
-			uninstalledAddOnsTable.getColumnModel().getColumn(4).setPreferredWidth(60);
-			uninstalledAddOnsTable.getColumnModel().getColumn(5).setPreferredWidth(40);
-
+			uninstalledAddOnsTable.getColumnModel().getColumn(0).setMaxWidth(20);//Icon
+			uninstalledAddOnsTable.getColumnExt(0).setSortable(false); //Icon doesn't need sorting
+			uninstalledAddOnsTable.getColumnModel().getColumn(1).setPreferredWidth(50);//Status
+			uninstalledAddOnsTable.getColumnModel().getColumn(2).setPreferredWidth(150);//Name
+			uninstalledAddOnsTable.getColumnModel().getColumn(3).setPreferredWidth(300);//Description
+			uninstalledAddOnsTable.getColumnExt(3).setSortable(false);//Description doesn't need sorting
+			uninstalledAddOnsTable.getColumnModel().getColumn(4).setPreferredWidth(60);//Update (version number)
+			uninstalledAddOnsTable.getColumnExt(4).setSortable(false);//Update doesn't need sorting
+			uninstalledAddOnsTable.getColumnModel().getColumn(5).setPreferredWidth(40);//Checkbox
+			uninstalledAddOnsTable.getColumnExt(5).setSortable(false);//Checkbox doesn't need sorting
+
+
+			uninstalledAddOnsTable.setAutoCreateRowSorter(true);
+           
+            //Default sort by status (column 1) descending (Release, Beta, Alpha), and name (column 2) ascending 
+            List<RowSorter.SortKey> sortKeys = new ArrayList<RowSorter.SortKey>(2);
+            sortKeys.add(new RowSorter.SortKey(1, SortOrder.DESCENDING));

```

Comment:
But it's improving the code base, so that's not a bad thing, right?

> which view are you talking about?

I was talking about the View class. [1]
Maybe there's a better place for that?

> Am I looping through the AddOn.Status enum to get all the possible statuses? 

Yeah, that's fine (but we should verify that the internationalised name for the Status(es) exist, otherwise it might start failing if a new Status is added without the corresponding internationalised name).

> How can I getStatusUI based solely on AddOn.Status where am I constructing it so that the internationalized stringRepresentation has been passed? 

I was thinking on the View class to hold a `Map<AddOn.Status, StatusUI>`, so the method "getStatusUI(AddOn.Status)" would just do a get on the map and return the StatusUI (which was previously constructed).


> null handling has been added to the constructor, it defaults to use of AddOn.Status.unknown if necessary.

OK, how about the string representation? Which value are you setting in that case?


[1] https://github.com/zaproxy/zaproxy/blob/61dab8de452dbe445e956484c085d4ec5384cdd0/src/org/parosproxy/paros/view/View.java